elytron/WFCORE-5744-ssl-certificate-expiration-warning.adoc

```diff
@@ -0,0 +1,218 @@
+ - elytron
+# Specify the stability level of the feature.
+# Values can be one of: experimental preview community default
+stability-level:
+ - Community
+issue:
+ - https://github.com/wildfly/wildfly-proposals/issues/685
+feature-team:
+ developer: bbaranow@redhat.com
+ sme:
+  - dlofthou@redhat.com
+ outside-perspective:
+  - frainone@redhat.com
+  - ropalka@redhat.com
+# If this issue tracks the promotion to a higher stability level of a previously
+# completed feature, provide the URL of the https://github.com/wildfly/wildfly-proposals/issues
+# issue that was used to track the previous feature.
+promotes:
+# This should be blank during initial development of a feature. It may be used
+# after the feature is completed if a subsequent issue is field to track promotion
+# of this feature to a higher stability level
+promoted-by:
+---
+= Display warning for expiring ssl ceritificates
+:author:            Bartosz Baranowski
+:email:             bbaranow@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+__<The entire document should be one to two pages long. We will write each analysis document as if it is a conversation with a future developer. This requires a good writing style, with full sentences organized into paragraphs. Bullets are acceptable only for visual style, not as an excuse for writing sentence fragments.>__
+
+== Overview
+
+Goal of this RFE is two fold, to formalize and improve ways to get feedback on state of SSL certificates in server and to add periodic check, based on configuration.
+
+Periodic check is more or less self explanatory. Currently upon startup elytron subsystem will will print out information about certificates that expired. There are few problems with this. Firstly its exception driven and lacks a bit of information( ie keystore). Secondly its one-off on startup, this operation potentially should be performed periodically.
+
+Secondly currently there exist dedicated command _should-renew-certificate(date_date_to_check)_ . It would be good to provide such detail along with _read-alias/s_ command output.
+The _read-alias_ will will provide health indication with fixed TTL( of 7 days at least), while _should-renew-certificate_ will allow detailed inerogration on certificate health.
+
+
+=== User Stories
+
+Long runing server, with periodic information on certificates will allow admins to not only setup tools to warn them about potential security risk, but also schedule maintanence window based on this.
+
+== Issue Metadata
+
+
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1863[EAP7-1863]
+* https://issues.redhat.com/browse/WFCORE-5744[WFCORE-5744]
+
+=== Affected Projects or Components
+
+ * Elytron integration
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
```

Comment:
There should be a requirement(s) around administrative RBAC controls related to this.

This is something we've probably missed in a lot of analyses, so I'm being a bit unfair asking for it here, but we should start making it a practice to cover this and thus hopefully ensure the code does too.

@darranl Do you think this is something to mention in the template comment in 'Security Considerations'? The info would naturally be Requirement items, but the suggestion comment more naturally fits in security considerations.

Reply:
For this proposal I think it should be an ACK that the Elytron subsystem has RBAC defined at the root - but +1 I think the relationship to RBAC would be good to call out and test in the default template.

Reply:
https://github.com/wildfly/wildfly-proposals/pull/726

Reply:
@darranl so no need to include this in proposal?